jme3-examples/src/main/java/jme3test/terrain/PBRTerrainAdvancedTest.java

```diff
@@ -0,0 +1,428 @@
+   //     TerrainLodControl control = new TerrainLodControl(terrain, getCamera());
+  //      control.setLodCalculator(new DistanceLodCalculator(65, 2.7f)); // patch size, and a multiplier
+     //   terrain.addControl(control);
+        terrain.setMaterial(matTerrain);
+        terrain.setModelBound(new BoundingBox());
+        terrain.updateModelBound();
+        terrain.setLocalTranslation(0, -100, 0);
+        terrain.setLocalScale(1f, 1f, 1f);
+        rootNode.attachChild(terrain);
+
+        Node probeNode = (Node) assetManager.loadModel("Scenes/lightprobe/quarry_Probe.j3o");  
+        
+        LightProbe probe = (LightProbe) probeNode.getLocalLightList().iterator().next();
+        probe.setPosition(new Vector3f(0, 0, 0));
+        
+        probe.setBounds(new BoundingSphere(5000, probe.getPosition()));
+        
+        rootNode.addLight(probe);
+        
+        directionalLight = new DirectionalLight();
+        directionalLight.setDirection((new Vector3f(-0.3f, -0.5f, -0.3f)).normalize());
+        directionalLight.setColor(ColorRGBA.White);
+        rootNode.addLight(directionalLight);
+        
+        ambientLight = new AmbientLight();
+        directionalLight.setColor(ColorRGBA.White);
+        rootNode.addLight(ambientLight);
+        
+
+        cam.setLocation(new Vector3f(0, 10, -10));
+        cam.lookAtDirection(new Vector3f(0, -1.5f, -1).normalizeLocal(), Vector3f.UNIT_Y);
+       
+        
+    }
+
+    private void setupKeys() {
+        flyCam.setMoveSpeed(50);
+        inputManager.addMapping("triPlanar", new KeyTrigger(KeyInput.KEY_P));
+        inputManager.addMapping("toggleNight", new KeyTrigger(KeyInput.KEY_N));
+        
+        inputManager.addListener(actionListener, "triPlanar");
+        inputManager.addListener(actionListener, "toggleNight");
+        
+        
+        keybindingsText = new BitmapText(assetManager.loadFont("Interface/Fonts/Default.fnt"));
+        keybindingsText.setText("Press 'N' to toggle day/night fade (takes a moment) \nPress 'P' to toggle tri-planar mode");
+        getGuiNode().attachChild(keybindingsText);
+        keybindingsText.move(new Vector3f(200,120,0));
+        
+    }
+    private ActionListener actionListener = new ActionListener() {
+
+        public void onAction(String name, boolean pressed, float tpf) {
+            if (name.equals("triPlanar") && !pressed) {
+                triPlanar = !triPlanar;
+                if (triPlanar) {
+                    matTerrain.setBoolean("useTriPlanarMapping", true);
+                    // planar textures don't use the mesh's texture coordinates but real world coordinates,
+                    // so we need to convert these texture coordinate scales into real world scales so it looks
+                    // the same when we switch to/from tr-planar mode (1024f is the alphamap size)
+                    matTerrain.setFloat("AlbedoMap_0_scale", 1f / (float) (1024f / dirtScale));
```

Comment:
Casting to float is unnecessary here and (in the following statements) because float divided-by float always yields a float. Please remove the casts.